packages/eslint-plugin/tests/rules/restrict-plus-operands.test.ts

```diff
@@ -411,6 +431,90 @@ function foo<T extends 1>(a: T) {
         },
       ],
     },
+    {
+      code: `
+        declare const a: boolean & string;
+        declare const b: string;
+        const x = a + b;
+      `,
+      errors: [
+        {
+          messageId: 'notStrings',
+          line: 4,
+          column: 19,
+        },
+      ],
+    },
+    {
+      code: `
+        declare const a: number & string;
+        declare const b: string;
+        const x = a + b;
+      `,
+      errors: [
+        {
+          messageId: 'notStrings',
+          line: 4,
+          column: 19,
+        },
+      ],
+    },
+    {
+      code: `
+        declare const a: symbol & string;
+        declare const b: string;
+        const x = a + b;
+      `,
+      errors: [
+        {
+          messageId: 'notStrings',
+          line: 4,
+          column: 19,
+        },
+      ],
+    },
+    {
+      code: `
+        declare const a: object & string;
+        declare const b: string;
+        const x = a + b;
+      `,
+      errors: [
+        {
+          messageId: 'notStrings',
+          line: 4,
+          column: 19,
+        },
+      ],
+    },
+    {
+      code: `
+        declare const a: never & string;
+        declare const b: string;
+        const x = a + b;
+      `,
+      errors: [
+        {
+          messageId: 'notStrings',
+          line: 4,
+          column: 19,
+        },
+      ],
+    },
+    {
+      code: `
+        declare const a: any & string;
+        declare const b: string;
+        const x = a + b;
+      `,
+      errors: [
+        {
+          messageId: 'notStrings',
+          line: 4,
+          column: 19,
+        },
+      ],
+    },
```

Comment:
You will need a test case with an intersection that doesn't contain a string for the code coverage to be happy:
```suggestion
    },
    {
       code: `
         declare const a: {a: 1} & {b: 2};
         declare const b: string;
         const x = a + b;
       `,
       errors: [
         {
           messageId: 'notStrings',
           line: 4,
           column: 19,
         },
       ],
     },
```